ui/apps/dashboard/src/app/(organization-active)/(dashboard)/billing/usage/page.tsx

```diff
@@ -1,39 +1,138 @@
 'use client';
 
+import { useState } from 'react';
+import dynamic from 'next/dynamic';
+import Link from 'next/link';
+import { Select, type Option } from '@inngest/components/Select/Select';
+import ToggleGroup from '@inngest/components/ToggleGroup/ToggleGroup';
 import { useQuery } from 'urql';
 
+import UsageMetadata from '@/components/Billing/Usage/Metadata';
 import { graphql } from '@/gql';
-import LoadingIcon from '@/icons/LoadingIcon';
-import { BillableStepUsage } from '../../settings/billing/BillableStepUsage/BillableStepUsage';
+import { pathCreator } from '@/utils/urls';
+
+const BillableUsageChart = dynamic(() => import('@/components/Billing/Usage/BillableUsageChart'), {
+  ssr: false,
+});
 
 const GetBillingInfoDocument = graphql(`
   query GetBillingInfo {
     account {
-      plan {
-        id
-        features
+      entitlements {
+        stepCount {
+          usage
+          limit
+        }
+        runCount {
+          usage
+          limit
+        }
       }
     }
   }
 `);
 
-export default function Billing() {
+type Period = Option & { id: 'current' | 'previous' };
+
+const options: Period[] = [
+  {
+    id: 'current',
+    name: 'This month',
+  },
+  {
+    id: 'previous',
+    name: 'Last month',
+  },
+];
+
+export default function Billing({
+  searchParams: { previous },
+}: {
+  searchParams: { previous: boolean };
+}) {
   const [{ data, fetching }] = useQuery({
     query: GetBillingInfoDocument,
   });
 
-  if (!data || fetching) {
-    return (
-      <div className="flex h-full min-h-[297px] w-full items-center justify-center overflow-hidden">
-        <LoadingIcon />
-      </div>
-    );
-  }
+  const [currentPage, setCurrentPage] = useState('step');
+  const [selectedPeriod, setSelectedPeriod] = useState<Period>(
+    previous ? options[1]! : options[0]!
```

Comment:
The `!` operator is a type cast so we should avoid it. You don't need it if you make `options` a tuple instead of an array, like this:

```ts
const options = [
  {
    id: 'current',
    name: 'This month',
  },
  {
    id: 'previous',
    name: 'Last month',
  },
] as const satisfies Period[];
```